src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -68,20 +68,40 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 			conptyInheritCursor: true
 		};
 
-		stat(shellLaunchConfig.executable!).then(stat => {
+		const cwdVerification = stat(cwd).then(stat => {
+			if (!stat.isDirectory()) {
+				return Promise.reject(SHELL_CWD_INVALID_EXIT_CODE);
+			}
+			return;
```

Comment:
I think these 4 returns at the end are redundant?

Reply:
They're needed so that it returns a resolved void promise, else typescript yells at me and says "Not all paths return a value"

Reply:
Does changing it to `async stat => {` fix that?

Reply:
It does because I think all async functions implicitly return promises. I have changed them to be async